rest/views.py

```diff
@@ -815,54 +789,34 @@ def post(self, request, *args, **kwargs):
         The export list will be left alone. Once the list is exported
         the client may DELETE the export list as a separate REST call.
         """
-
         repo_slug = self.kwargs['repo_slug']
-        try:
-            exports = set(self.request.session[EXPORTS_KEY][repo_slug])
-        except KeyError:
-            exports = set()
-
-        ids = self.request.data['ids']
-        for resource_id in ids:
-            if resource_id not in exports:
-                raise ValidationError("id {id} is not in export list".format(
-                    id=resource_id
-                ))
+        export_tasks = [
+            task for task in get_tasks(self.request.session).values()
+            if task['task_type'] == EXPORT_TASK_TYPE and
+            task['task_info']['repo_slug'] == repo_slug
+        ]
 
         # Cancel any old tasks.
-        old_tasks = self.request.session.get(
-            EXPORT_TASK_KEY, {}).get(repo_slug, {})
-        for task_id in old_tasks.keys():
-            AsyncResult(task_id).revoke()
-
-        # Clear task list.
-        if EXPORT_TASK_KEY not in self.request.session:
-            self.request.session[EXPORT_TASK_KEY] = {}
-        self.request.session[EXPORT_TASK_KEY][repo_slug] = {}
-
-        learning_resources = LearningResource.objects.filter(
-            id__in=ids).all()
-        result = export_resources.delay(
-            learning_resources, self.request.user.username)
-
-        if result.successful():
-            name, collision = result.get()
-            url = default_storage.url(name)
-        else:
-            collision = False
-            url = ""
-
-        # Put new task in session.
-        self.request.session[EXPORT_TASK_KEY][repo_slug][result.id] = {
-            "id": result.id,
-            "initial_state": result.state,
-            "url": url,
-            "collision": collision
-        }
-        self.request.session.modified = True
+        for task in export_tasks:
+            remove_task(self.request.session, task['id'])
```

Comment:
The user may have an export task for another repository in progress (unlikely but possible)